crypto-common/src/lib.rs

```diff
@@ -156,10 +159,17 @@ pub trait KeyInit: KeySizeUser + Sized {
             .map_err(|_| InvalidLength)
     }
 
+    /// Generate random key using the operating system's secure RNG.
+    #[cfg(feature = "getrandom")]
+    #[inline]
+    fn generate_key() -> Key<Self> {
+        Self::generate_key_with_rng(&mut OsRng)
+    }
+
     /// Generate random key using the provided [`CryptoRngCore`].
     #[cfg(feature = "rand_core")]
     #[inline]
-    fn generate_key(mut rng: impl CryptoRngCore) -> Key<Self> {
+    fn generate_key_with_rng(rng: &mut impl CryptoRngCore) -> Key<Self> {
         let mut key = Key::<Self>::default();
         rng.fill_bytes(&mut key);
```

Comment:
@newpavlov should we also make these methods fallible and call `try_fill_bytes`?

Reply:
It may be worth to add separate `try_*` variants and add a note to the `generate_*` method docs about potential (improbable) panics.